### ISSUE TITLE
stuck in the single-node rl training

### ISSUE THREAD

**Issue by g-makerr** [TIGER-AI-Lab/Pixel-Reasoner#26]
@HaozheH3 Hi, Haozhe.
I encounter some issues in the single-node rl training, I want to  find whether it is an environmental issue. Could you release your pip list.
My program seems to be stuck here.

absl-py                           2.3.0
accelerate                        1.8.1
aiohappyeyeballs                  2.6.1
aiohttp                           3.12.13
aiohttp-cors                      0.8.1
aiosignal                         1.3.2
airportsdata                      20250622
annotated-types                   0.7.0
antlr4-python3-runtime            4.13.2
anyio                             4.9.0
astor                             0.8.1
async-timeout                     5.0.1
attrs                             25.3.0
av                                14.4.0
bitsandbytes                      0.46.0
blake3                            1.0.5
blinker                           1.9.0
cachetools                        5.5.2
certifi                           2025.6.15
charset-normalizer                3.4.2
click                             8.2.1
cloudpickle                       3.1.1
colorful                          0.5.7
compressed-tensors                0.9.1
datasets                          3.6.0
deepspeed                         0.15.0
depyf                             0.18.0
dill                              0.3.8
diskcache                         5.6.3
distlib                           0.3.9
distro                            1.9.0
einops                            0.8.1
exceptiongroup                    1.3.0
fastapi                           0.115.14
filelock                          3.18.0
flash_attn                        2.8.0.post2
Flask                             3.1.1
frozenlist                        1.7.0
fsspec                            2025.3.0
gguf                              0.10.0
gitdb                             4.0.12
GitPython                         3.1.44
google-api-core                   2.25.1
google-auth                       2.40.3
googleapis-common-protos          1.70.0
grpcio                            1.73.1
h11                               0.16.0
hf-xet                            1.1.5
hjson                             3.1.0
httpcore                          1.0.9
httptools                         0.6.4
httpx                             0.28.1
huggingface-hub                   0.33.1
idna                              3.10
importlib_metadata                8.7.0
interegular                       0.3.3
isort                             6.0.1
itsdangerous                      2.2.0
Jinja2                            3.1.6
jiter                             0.10.0
jsonlines                         4.0.0
jsonschema                        4.24.0
jsonschema-specifications         2025.4.1
lark                              1.2.2
latex2sympy2_extended             1.10.1
Levenshtein                       0.27.1
lightning-utilities               0.14.3
lm-format-enforcer                0.10.11
loralib                           0.1.2
Markdown                          3.8.2
MarkupSafe                        3.0.2
math-verify                       0.7.0
mistral_common                    1.6.3
mpmath                            1.3.0
msgpack                           1.1.1
msgspec                           0.19.0
multidict                         6.6.3
multiprocess                      0.70.16
nest-asyncio                      1.6.0
networkx                          3.4.2
ninja                             1.11.1.4
numpy                             1.26.4
nvidia-cublas-cu12                12.4.5.8
nvidia-cuda-cupti-cu12            12.4.127
nvidia-cuda-nvrtc-cu12            12.4.127
nvidia-cuda-runtime-cu12          12.4.127
nvidia-cudnn-cu12                 9.1.0.70
nvidia-cufft-cu12                 11.2.1.3
nvidia-cufile-cu12                1.11.1.6
nvidia-curand-cu12                10.3.5.147
nvidia-cusolver-cu12              11.6.1.9
nvidia-cusparse-cu12              12.3.1.170
nvidia-cusparselt-cu12            0.6.3
nvidia-ml-py                      12.575.51
nvidia-nccl-cu12                  2.21.5
nvidia-nvjitlink-cu12             12.4.127
nvidia-nvtx-cu12                  12.4.127
openai                            1.93.0
opencensus                        0.11.4
opencensus-context                0.1.3
opencv-python-headless            4.11.0.86
openrlhf                          0.6.0         /path/Pixel-Reasoner/curiosity_driven_rl
optimum                           1.26.1
outlines                          0.1.11
outlines_core                     0.1.26
packaging                         25.0
pandas                            2.3.0
partial-json-parser               0.2.1.1.post6
peft                              0.15.2
pillow                            11.3.0
pip                               25.1
platformdirs                      4.3.8
prometheus_client                 0.22.1
prometheus-fastapi-instrumentator 7.1.0
propcache                         0.3.2
proto-plus                        1.26.1
protobuf                          6.31.1
psutil                            7.0.0
py-cpuinfo                        9.0.0
py-spy                            0.4.0
pyarrow                           20.0.0
pyasn1                            0.6.1
pyasn1_modules                    0.4.2
pycountry                         24.6.1
pydantic                          2.11.7
pydantic_core                     2.33.2
pynvml                            12.0.0
python-dateutil                   2.9.0.post0
python-dotenv                     1.1.1
pytz                              2025.2
PyYAML                            6.0.2
pyzmq                             27.0.0
qwen-vl-utils                     0.0.11
RapidFuzz                         3.13.0
ray                               2.42.0
referencing                       0.36.2
regex                             2024.11.6
requests                          2.32.4
rpds-py                           0.26.0
rsa                               4.9.1
safetensors                       0.5.3
sentencepiece                     0.2.0
sentry-sdk                        2.32.0
setuptools                        78.1.1
six                               1.17.0
smart_open                        7.3.0
smmap                             5.0.2
sniffio                           1.3.1
starlette                         0.46.2
sympy                             1.13.1
tensorboard                       2.19.0
tensorboard-data-server           0.7.2
tiktoken                          0.9.0
tokenizers                        0.21.2
torch                             2.5.1
torchaudio                        2.5.1
torchmetrics                      1.7.3
torchvision                       0.20.1
tqdm                              4.67.1
transformers                      4.49.0.dev0
transformers-stream-generator     0.0.5
triton                            3.1.0
typing_extensions                 4.14.0
typing-inspection                 0.4.1
tzdata                            2025.2
urllib3                           2.5.0
uvicorn                           0.35.0
uvloop                            0.21.0
virtualenv                        20.31.2
vllm                              0.7.2
wandb                             0.21.0
watchfiles                        1.1.0
websockets                        15.0.1
Werkzeug                          3.1.3
wheel                             0.45.1
wrapt                             1.17.2
xformers                          0.0.28.post3
xgrammar                          0.1.20
xxhash                            3.5.0
yarl                              1.20.1
zipp                              3.23.0

<img width="1177" alt="Image" src="https://github.com/user-attachments/assets/ab7cf397-2073-4347-8037-e79aa2e19dde" />

**Comment by g-makerr** [comment]
ray job submit --address="http://127.0.0.1:8265" \
--runtime-env-json="$RUNTIME_ENV_JSON" \
-- python3 -m openrlhf.cli.train_ppo_ray \
--vllm_enable_sleep \
--vllm_gpu_memory_utilization 0.85 \
--vllm_sync_backend gloo \
--pretrain $PRETRAIN_MODEL \
--save_path $SAVE_PATH \
--n_samples_per_prompt ${nsamples} \
--max_epochs 1 \
--num_episodes ${nepoch} \
--filter ${filter} \
--prompt_max_len 2048 \
--max_out_tokens ${maxtoken} \
--max_samples 100000 \
--generate_max_len ${maxlen} \
--advantage_estimator ${algo} \
--zero_stage 3 \
--controlled_shuffle ${repeat} \
--bf16 \
--actor_learning_rate ${lr}e-7 \
--rule_reward ${rule_reward} \
--temperature 1.0 \
--val_temperature 0.6 \
--top_p 0.95 \
--training_mode ${mode} \
--init_kl_coef 0.0 \
--aux_loss_coef ${aux} \
--entropy_loss_coef 0.0 \
--prompt_data $DATASET \
--input_key question \
--apply_chat_template \
--normalize_reward \
--flash_attn \
--gradient_checkpointing \
--ckpt_path $SAVE_PATH \
--save_steps 3 \
--eval_steps ${evalsteps} \
--max_ckpt_num 3 \
--save_hf_ckpt \
--disable_ds_ckpt \
--disable_fast_tokenizer \
--wandb_run_name $save_name \
--system_prompt ${sys} \
--use_kl_estimator_k3 \
--wandb_project vlm-rl \
--buffer_norm 0 \
--train_vlm \
--filter ${filter} \
--data_version ${dataver} \
--loss_version ${lossver} \
--format ${fmt} \
${post_args[@]} 
   # --train_vlm

**Comment by HaozheH3** [comment]
The screenshot reveals not much information. 
I think the init_process_group() is not successful, it could be due to improper NCCL configs. 
Have you trained with torchrun or deepspeed before?
Can you send the full logs to jasper.whz@outlook.com?

**Comment by g-makerr** [comment]
Thanks for your reply, I have solved this problem. I just restarted my server and the problem disappeared. I change transformers==4.49.0 to 4.49.0.dev0 for issue [https://github.com/vllm-project/vllm/issues/13285](url) and modified the modeling_qwen2_5_vl.py for issue. [https://github.com/modelscope/ms-swift/issues/3156](url).
Now I can start the rl training well.


> The screenshot reveals not much information. I think the init_process_group() is not successful, it could be due to improper NCCL configs. Have you trained with torchrun or deepspeed before? Can you send the full logs to [jasper.whz@outlook.com](mailto:jasper.whz@outlook.com)?

**Comment by HaozheH3** [comment]
Glad to hear! Then I will close this.
